packages/runtime/runtime-utils/src/dataStoreRegistry.ts

```diff
@@ -0,0 +1,62 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import {
+    FluidDataStoreRegistryEntry,
+    IFluidDataStoreRegistry,
+    FluidDataStoreRegistry,
+    FluidDataStoreRegistryEntries,
+    IProvideFluidDataStoreRegistry,
+} from "@fluidframework/runtime-definitions";
+
+export function createDataStoreRegistry(arg: FluidDataStoreRegistry): IFluidDataStoreRegistry {
+    // Ensure that FluidDataStoreRegistry type did not change from original
+    // We do casting below and need to ensure that only these two types can come in
+    const entries: FluidDataStoreRegistryEntries | IFluidDataStoreRegistry = arg;
+
+    const registry = (entries as IProvideFluidDataStoreRegistry).IFluidDataStoreRegistry;
+    if (registry !== undefined) {
+        return registry;
+    }
+    return new DataStoreRegistry(entries as FluidDataStoreRegistryEntries);
+}
+
+class DataStoreRegistry implements IFluidDataStoreRegistry {
```

Comment:
Do we have any testing to make sure that subcomponent registries still function?

Reply:
Yes, a lot of tests would fail, including jest tests of end-to-end experiences (like Pond).
I think the only interesting missing case is that createDataStoreRegistry() preserves the object if it provides registry - I'll add that.